Creating-Interactions.md

```diff
@@ -1,62 +1,158 @@
-Interactions provide a way for learners to submit answers to Oppia. They are implemented using custom HTML tags built with [AngularJS directives](http://seanhess.github.io/2013/10/14/angularjs-directive-design.html). Here's an example of such a tag:
+## Table of Contents
 
+* [Introduction](#introduction)
+* [Directory structure](#directory-structure)
+* [The `directives` folder](#the-directives-folder)
+  * [Components](#components)
+  * [Services](#services)
+* [Add a new Interaction](#add-a-new-interaction)
+  * [Testing](#testing)
+
+## Introduction
+
+Interactions provide a way for learners to submit answers to Oppia. They are implemented using custom HTML tags built with Angular Components. Here's an example of such a tag:
+
+```html
+<oppia-interactive-text-input></oppia-interactive-text-input>
 ```
-  <oppia-interactive-text-input></oppia-interactive-text-input>
-```
 
-In the Oppia codebase, interactions live in the `extensions/interactions` directory. Each interaction is made up of four main components:
+## Directory structure
+
+In the Oppia codebase, interactions live in the `extensions/interactions` directory. Each interaction's foler has the following structure:
+
+* A Python configuration file
+* An empty `__init__.py` file
+* A `directives` folder that contains Angular components that make up
+  the interaction. Each component is defined by a triplet of files: a
+  `*.html` file, a `*.ts` file, and a `*.spec.ts` file. This folder may
+  also contain `*.service.ts` files that define services, for example a
+  validation service.
+* A `*.ts` file that imports all the components from the `directives`
+  folder.
+* A `*.module.ts` file that defines the interaction as a module.
+
+## The directives folder
+
+### Components
+
+Each interaction needs the following components in its `directives` folder:
+
+* Interaction component: A form the learner will submit when responding to the interaction. For the Multiple Choice Input interaction, this form shows answer options for the user to select.
+* Response component: Displays the learner's response to the interaction. For the Multiple Choice Input interaction, this would be the user's selected answer.
+* Short response component: Displays a shortened form of the learner's response. For some interactions, this shortened form is the same as the response component.
```

Comment:
Maybe show a picture showing how this looks like in the learner view so that the reader can see the visual difference between the "response" and "short response" components.

Reply:
done